src/SonarScanner.MSBuild.PreProcessor/WebServer/SonarQubeWebServer.cs

```diff
@@ -45,16 +45,23 @@ public override async Task<bool> IsServerLicenseValid()
         {
             logger.LogDebug(Resources.MSG_CheckingLicenseValidity);
             var uri = GetUri("api/editions/is_valid_license");
-            var response = await downloader.TryGetLicenseInformation(uri);
-            var content = await response.Content.ReadAsStringAsync();
-            var json = JObject.Parse(content);
+            var response = await downloader.DownloadResource(uri);
+            if (response.StatusCode == HttpStatusCode.Unauthorized)
+            {
+                logger.LogError(Resources.ERR_InvalidCredentials);
+                return false;
+            }
+
+            var json = JObject.Parse(await response.Content.ReadAsStringAsync());
             if (response.StatusCode == HttpStatusCode.NotFound)
             {
+                // On other editions than community, if a license was not set, the response is: {"errors":[{"msg":"License not found"}]} and http status code 404 (not found).
```

Comment:
(follow-up of previous comment )

I got confused because you explicitly mention above `and http status code 404 (not found).` 

But the if above has `if (response.StatusCode == HttpStatusCode.NotFound)` makes it clear it's 404, so this info is a bit redundant in the comment.

Having the comments for editions totally makes sense